shared-data/js/__tests__/labwareDefSchemaV3.test.ts

```diff
@@ -61,19 +61,24 @@ const checkExtents = (labwareDef: LabwareDefinition3): void => {
     expect(backLeftBottom.y).toBeGreaterThanOrEqual(frontRightTop.y)
     expect(backLeftBottom.z).toBeLessThanOrEqual(frontRightTop.z)
   })
-  test('extents.footprint should be oriented correctly', () => {
-    const {
-      footprint: { backLeft, frontRight },
-    } = labwareDef.extents
-    expect(backLeft.x).toBeLessThanOrEqual(frontRight.x)
-    expect(backLeft.y).toBeGreaterThanOrEqual(frontRight.y)
+  test('slotFootprintAsChild feature should be oriented correctly', () => {
+    if (labwareDef.features.slotFootprintAsChild) {
+      const { backLeft, frontRight } = labwareDef.features.slotFootprintAsChild
+      expect(backLeft.x).toBeLessThanOrEqual(frontRight.x)
+      expect(backLeft.y).toBeGreaterThanOrEqual(frontRight.y)
+    }
   })
   test('extents.footprint should be contained inside extents.total', () => {
-    const { footprint, total } = labwareDef.extents
-    expect(footprint.backLeft.x).toBeGreaterThanOrEqual(total.backLeftBottom.x)
-    expect(footprint.backLeft.y).toBeLessThanOrEqual(total.backLeftBottom.y)
-    expect(footprint.frontRight.x).toBeLessThanOrEqual(total.frontRightTop.x)
-    expect(footprint.frontRight.y).toBeGreaterThanOrEqual(total.frontRightTop.y)
+    if (labwareDef.features.slotFootprintAsChild) {
+      const { backLeft, frontRight } = labwareDef.features.slotFootprintAsChild
+      const {
+        total: { backLeftBottom, frontRightTop },
+      } = labwareDef.extents
+      expect(backLeft.x).toBeGreaterThanOrEqual(backLeftBottom.x)
+      expect(backLeft.y).toBeLessThanOrEqual(backLeftBottom.y)
+      expect(frontRight.x).toBeLessThanOrEqual(frontRightTop.x)
+      expect(frontRight.y).toBeGreaterThanOrEqual(frontRightTop.y)
+    }
```

Comment:
Oh, we can check `z` here too, now that the footprint has it.

```suggestion
      const { backLeft, frontRight, z } = labwareDef.features.slotFootprintAsChild
      const {
        total: { backLeftBottom, frontRightTop },
      } = labwareDef.extents
      expect(backLeft.x).toBeGreaterThanOrEqual(backLeftBottom.x)
      expect(backLeft.y).toBeLessThanOrEqual(backLeftBottom.y)
      expect(frontRight.x).toBeLessThanOrEqual(frontRightTop.x)
      expect(frontRight.y).toBeGreaterThanOrEqual(frontRightTop.y)
      expect(z).toBeGreaterThanOrEqual(backLeftBottom.z)
      expect(z).toBeLessThanOrEqual(frontRightTop.z)
    }
```